content/numerical/IntegrateAdaptive.h

```diff
@@ -15,18 +15,21 @@
 #pragma once
 
 typedef double d;
-d simpson(d (*f)(d), d a, d b) {
+template<class F>
+d simpson(F f, d a, d b) {
 	d c = (a+b) / 2;
 	return (f(a) + 4*f(c) + f(b)) * (b-a) / 6;
 }
```

Comment:
We can shorten this a bit (and in particular get rid of the `template<class T>` bit) by making the function into a macro:
```
#define S(a,b) (f(a) + 4*f((a+b) / 2) + f(b)) * (b-a) / 6
```